service/efsPort.py

```diff
@@ -0,0 +1,630 @@
+        multipliers["droneBandwidth"] = round(getMaxRatio(preTraitMultipliers, postTraitMultipliers, "drones"), 6)
+        Fit.getInstance().recalc(fit)
+        return multipliers
+
+    @staticmethod
+    def getShipSize(groupID):
+        # Size groupings are somewhat arbitrary but allow for a more managable number of top level groupings in a tree structure.
+        frigateGroupNames = ["Frigate", "Shuttle", "Corvette", "Assault Frigate", "Covert Ops", "Interceptor",
+                             "Stealth Bomber", "Electronic Attack Ship", "Expedition Frigate", "Logistics Frigate"]
+        destroyerGroupNames = ["Destroyer", "Interdictor", "Tactical Destroyer", "Command Destroyer"]
+        cruiserGroupNames = ["Cruiser", "Heavy Assault Cruiser", "Logistics", "Force Recon Ship",
+                             "Heavy Interdiction Cruiser", "Combat Recon Ship", "Strategic Cruiser"]
+        bcGroupNames = ["Combat Battlecruiser", "Command Ship", "Attack Battlecruiser"]
+        bsGroupNames = ["Battleship", "Elite Battleship", "Black Ops", "Marauder"]
+        capitalGroupNames = ["Titan", "Dreadnought", "Freighter", "Carrier", "Supercarrier",
+                             "Capital Industrial Ship", "Jump Freighter", "Force Auxiliary"]
+        indyGroupNames = ["Industrial", "Deep Space Transport", "Blockade Runner",
+                          "Mining Barge", "Exhumer", "Industrial Command Ship"]
+        miscGroupNames = ["Capsule", "Prototype Exploration Ship"]
+        shipSizes = [
+            {"name": "Frigate", "groupIDs": map(lambda s: getGroup(s).ID, frigateGroupNames)},
+            {"name": "Destroyer", "groupIDs": map(lambda s: getGroup(s).ID, destroyerGroupNames)},
+            {"name": "Cruiser", "groupIDs": map(lambda s: getGroup(s).ID, cruiserGroupNames)},
+            {"name": "Battlecruiser", "groupIDs": map(lambda s: getGroup(s).ID, bcGroupNames)},
+            {"name": "Battleship", "groupIDs": map(lambda s: getGroup(s).ID, bsGroupNames)},
+            {"name": "Capital", "groupIDs": map(lambda s: getGroup(s).ID, capitalGroupNames)},
+            {"name": "Industrial", "groupIDs": map(lambda s: getGroup(s).ID, indyGroupNames)},
+            {"name": "Misc", "groupIDs": map(lambda s: getGroup(s).ID, miscGroupNames)}
+        ]
+        for size in shipSizes:
+            if groupID in size["groupIDs"]:
+                return size["name"]
+        sizeNotFoundMsg = "ShipSize not found for groupID: " + str(groupID)
+        return sizeNotFoundMsg
+
+    @staticmethod
+    def exportEfs(fit, groupID):
+        includeShipTypeData = groupID > 0
+        fitID = fit.ID
+        if includeShipTypeData:
+            fitName = fit.name
+        else:
+            fitName = fit.ship.name + ": " + fit.name
+        pyfalog.info("Creating Eve Fleet Simulator data for: " + fit.name)
+        sFit = Fit.getInstance()
+        sFit.recalc(fit)
```

Comment:
Don't think you need to recalc... once you get to this spot, the fit should have already been loaded and calc'd in pyfa. That being said, it doesn't really hurt.

Reply:
Same situation as create_all(), I'll move it out of EfsPort.